classes/services/queryBuilders/PKPUserQueryBuilder.inc.php

```diff
@@ -454,6 +472,11 @@ public function getQuery() {
 			$q->whereIn('ug.role_id', $this->roleIds);
 		}
 
+		// user group
+		if (!is_null($this->userGroupIds)) {
```

Comment:
(If you move away from null `userGroupIds` towards empty arrays, as I mentioned above, you'd need to adjust this to `!empty` -- but again, bikeshedding.)